OSDOCS-8615: Configuring PID limits (#69352)
openshift/openshift-docs: modules/setting-higher-pid-limit-on-existing-cluster.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-configuring-pid-limits.adoc

:_mod-docs-content-type: PROCEDURE
[id="setting-higher-pid-limit-on-existing-cluster_{context}"]
= Setting a higher PID limit on an existing {product-title} cluster

> You can set a higher `podPidsLimit` on an existing {product-title} cluster by creating or editing a kubeletconfig that changes the `--pod-pids-limit` parameter.

Reviewer: I'd also suggest to include a guidance to use status of machine config pool to ensure that the reboot is completed? The instruction could either be linked to https://docs.openshift.com/container-platform/4.8/post_installation_configuration/machine-configuration-tasks.html#checking-mco-status_post-install-machine-configuration-tasks or restated for better UX.

[IMPORTANT]
====

> Changing the `podPidsLimit` on an existing cluster will trigger non-control plane nodes in the cluster to reboot one at a time. Red Hat recommends making this change outside of peak usage hours for your cluster, and avoiding upgrading or hibernating your cluster until all nodes have rebooted.

Reviewer 2: ```suggestion
Changing the `podPidsLimit` on an existing cluster will trigger non-control plane nodes in the cluster to reboot one at a time. Make this change outside of peak usage hours for your cluster and avoid upgrading or hibernating your cluster until all nodes have rebooted.
```
Per the [SSG](https://redhat-documentation.github.io/supplementary-style-guide/#recommend), avoid "recommends" completely.

'Instead of "Red Hat recommends", direct users to take the recommended action. This allows Red Hat to be more prescriptive in documentation and prevent any user uncertainty, and is easier for upstream or downstream coordinated efforts.'

====

.Prerequisites

* You have logged in to your Red Hat account by using the ROSA CLI.
* You have installed the OpenShift CLI (`oc`).

.Procedure

* Create or edit the kubeletconfig to change the PID limit.
+
--

> ** If this is the first time you are changing the default PID limit, create the kubeletconfig and set the `--pod-pids-limit` value. 

Reviewer 2: ```suggestion
** If this is the first time you are changing the default PID limit, create the kubeletconfig and set the `--pod-pids-limit` value by running the following command: 
```
- Follow OCP guidelines for introducing a terminal command: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

+
[source,terminal]
----
$ rosa create kubeletconfig -c <cluster_name> --pod-pids-limit=<value>
----
+

> For example, to set a maximum of 16,384 PIDs per pod for cluster my-cluster when you have never changed the default value:

Reviewer 3: ```suggestion
For example, to set a maximum of 16,384 PIDs per pod for cluster my-cluster when you have never created a kubeletconfig
```

Reviewer 2: ```suggestion
For example, to set a maximum of 16,384 PIDs per pod for a cluster named `my-cluster` when you have never changed the default value, run the following command:
```
Revise this introductory phrase to be a complete sentence and to introduce the command using the [OCP docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command).

+
[source,terminal]
----
$ rosa create kubeletconfig -c my-cluster --pod-pids-limit=16384
----

> ** If you previously changed the default PID limit, edit the existing kubeletconfig and set the `--pod-pids-limit` value.

Reviewer 3: ```suggestion
** If you previously created a kubeletconfig, edit the existing kubeletconfig and set the `--pod-pids-limit` value.
```

> ** If you previously changed the default PID limit, edit the existing kubeletconfig and set the `--pod-pids-limit` value.

Reviewer 2: ```suggestion
** If you previously changed the default PID limit, edit the existing `kubeletconfig` object and set the `--pod-pids-limit` value.
```
- `kubeletconfig` is a literal and needs a descriptive noun. I'm no sure if it's an object, a file, or a config map or what, though.

+
[source,terminal]
----
$ rosa edit kubeletconfig -c <cluster_name> --pod-pids-limit=<value>
----
--
+
A cluster-wide rolling reboot of worker nodes is triggered.

.Verification

When each node in the cluster has rebooted, you can verify that the new setting is in place.

> * Check the Pod Pids limit in the kubeletconfig:

Reviewer 2: ```suggestion
* Check the Pod Pids limit in the `kubeletconfig` [object???], by running the following command:
```
- Format `kubeletconfig` as a literal.
- Revise this introductory phrase to introduce the command using the [OCP docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command).

+
[source,terminal]
----
$ rosa describe kubeletconfig --cluster=<cluster_name>
----
+

> You should see the new Pod Pids limit in the output. For example:

Reviewer 2: Yes. @shipsing is correct. This line can be completely omitted. 

- For future reference, we need a complete sentence as an introduction. `For example:` by itself does not suffice.

```suggestion
The new Pod Pids limit appears in the output, as shown in the following example:
```
- Also for future reference, [per the ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s), avoid the word "should" and try to rewrite in a more direct way; "should" implies some doubt as to whether or not the action will have the desired result.
